slave.mk

```diff
@@ -0,0 +1,307 @@
+# Add new package for build:
+#     SOME_NEW_WHL = some_new_whl.whl
+#     $(SOME_NEW_WHL)_SRC_PATH = $(SRC_PATH)/project_name
+#     $(SOME_NEW_WHL)_PYTHON_VERSION = 2 (or 3)
+#     $(SOME_NEW_WHL)_DEPENDS = $(SOME_OTHER_WHL1) $(SOME_OTHER_WHL2) ...
+#     SONIC_PYTHON_WHEELS += $(SOME_NEW_DEB)
+$(addprefix $(PYTHON_WHEELS_PATH)/, $(SONIC_PYTHON_WHEELS)) : $(PYTHON_WHEELS_PATH)/% : .platform $$(addsuffix -install,$$(addprefix $(PYTHON_WHEELS_PATH)/,$$($$*_DEPENDS)))
+	$(HEADER)
+	pushd $($*_SRC_PATH) $(LOG)
+	python$($*_PYTHON_VERSION) setup.py bdist_wheel $(LOG) || exit 1
+	popd $(LOG)
+	mv $($*_SRC_PATH)/dist/$* $(PYTHON_WHEELS_PATH) $(LOG)
+	$(FOOTER)
+
+# Targets for installing python wheels.
+# Autogenerated
+SONIC_INSTALL_WHEELS = $(addsuffix -install, $(addprefix $(PYTHON_WHEELS_PATH)/, $(SONIC_PYTHON_WHEELS)))
+$(SONIC_INSTALL_WHEELS) : $(PYTHON_WHEELS_PATH)/%-install : .platform $$(addsuffix -install,$$(addprefix $(PYTHON_WHEELS_PATH)/,$$($$*_DEPENDS))) $(PYTHON_WHEELS_PATH)/$$*
+	$(HEADER)
+	[ -f $(PYTHON_WHEELS_PATH)/$* ] || { echo $(PYTHON_WHEELS_PATH)/$* does not exist $(LOG) && exit 1; }
+	# put a lock here because dpkg does not allow installing packages in parallel
+	while true; do
+	if mkdir $(PYTHON_WHEELS_PATH)/pip_lock &> /dev/null; then
+	{ sudo pip$($*_PYTHON_VERSION) install $(PYTHON_WHEELS_PATH)/$* $(LOG) && rm -d $(PYTHON_WHEELS_PATH)/pip_lock && break; } || { rm -d $(PYTHON_WHEELS_PATH)/pip_lock && exit 1 ; }
+	fi
+	done
+	$(FOOTER)
+
+###############################################################################
+## Docker images related targets
+###############################################################################
+
+# start docker daemon
+docker-start :
+	@sudo service docker start &> /dev/null && sleep 1
+
+# targets for building simple docker images that do not depend on any debian packages
+$(addprefix $(TARGET_PATH)/, $(SONIC_SIMPLE_DOCKER_IMAGES)) : $(TARGET_PATH)/%.gz : .platform docker-start $$(addsuffix -load,$$(addprefix $(TARGET_PATH)/,$$($$*.gz_LOAD_DOCKERS)))
+	$(HEADER)
+	docker build --no-cache -t $* $($*.gz_PATH) $(LOG) || { rm -f $@ && exit 1 ; }
+	docker save $* | gzip -c > $@
+	$(FOOTER)
+
+# Targets for building docker images
+$(addprefix $(TARGET_PATH)/, $(SONIC_DOCKER_IMAGES)) : $(TARGET_PATH)/%.gz : .platform docker-start $$(addprefix $(DEBS_PATH)/,$$($$*.gz_DEPENDS) $$($$*.gz_FILES)) $$(addprefix $(PYTHON_WHEELS_PATH)/,$$($$*.gz_PYTHON_WHEELS)) $$(addsuffix -load,$$(addprefix $(TARGET_PATH)/,$$($$*.gz_LOAD_DOCKERS)))
+	$(HEADER)
+	mkdir -p $($*.gz_PATH)/deps $(LOG)
+	mkdir -p $($*.gz_PATH)/python-wheels $(LOG)
+	sudo mount --bind $(DEBS_PATH) $($*.gz_PATH)/deps $(LOG)
+	sudo mount --bind $(PYTHON_WHEELS_PATH) $($*.gz_PATH)/python-wheels $(LOG)
+	sed 's/SED_DPKG/RUN cd deps \&\& dpkg -i $(shell printf "$(subst $(SPACE),\n,$(call expand,$($*.gz_DEPENDS),RDEPENDS))\n" | awk '!a[$$0]++')/g' $($*.gz_PATH)/Dockerfile.template > $($*.gz_PATH)/Dockerfile
```

Comment:
It is a common practice to use sed in composition with makefiles
In what way is sed not mature enough? I need to add a command into docker file based on variables from Makefile, and for this case sed looks like the most convenient tool